xsi:attributes break parsing.

If I have some XML which declares its own schema, which is useful sometimes, the XML gets unparsable.

E.g.:
```python
def test_tolerate_xsi(self):
       xml =  """<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<!-- This clip does not have a mid. So it will be a new clip when you post this -->
<program xmlns="urn:vpro:media:update:2009"
         type="CLIP" avType="VIDEO"
         publishStart="2012-01-11T16:16:01.287+01:00" embeddable="true"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="urn:vpro:media:update:2009 https://poms.omroep.nl/schema/update/vproMediaUpdate.xsd"        
         >
  <broadcaster>VPRO</broadcaster>
  <title type="MAIN">TEST TEST</title>
  <tag>Kabouters</tag>
  <ageRating>ALL</ageRating>
  <email>programma@avro.nl</email>  
</program>         
         """
       program_update =  parser.from_string(xml)
       self.assertEqual(program_update.ageRating, AgeRatingType.ALL)
``` 
The parser step fails with the following error:
```

self = <xsdata.formats.dataclass.parsers.nodes.element.ElementNode object at 0x12285a670>
params = {'avType': <AvTypeEnum.VIDEO: 'VIDEO'>, 'embeddable': True, 'publishStart': XmlDateTime(2012, 1, 11, 16, 16, 1, 287000000, 60), 'typeValue': <ProgramTypeEnum.CLIP: 'CLIP'>}

    def bind_attrs(self, params: Dict):
        """Parse the given element's attributes and any text content and return
        a dictionary of field names and values based on the given class
        metadata."""
    
        if not self.attrs:
            return
    
        for qname, value in self.attrs.items():
            var = self.meta.find_attribute(qname)
            if var and var.name not in params:
                self.bind_attr(params, var, value)
            else:
                var = self.meta.find_any_attributes(qname)
                if var:
                    self.bind_any_attr(params, var, qname, value)
                else:
                    if self.config.fail_on_unknown_attributes:
>                       raise ParserError(
                            f"Unknown attribute {self.meta.qname}:{qname}"
                        )
E                       xsdata.exceptions.ParserError: Unknown attribute {urn:vpro:media:update:2009}program:{http://www.w3.org/2001/XMLSchema-instance}schemaLocation

/Users/michiel/venvs/pyapi-env/lib/python3.11/site-packages/xsdata/formats/dataclass/parsers/nodes/element.py:138: ParserError

Process finished with exit code 1
```

I think this is erroneous. The attribute schemaLocation is not unknown, it is specified by xmlns:xsi. I think it can just be ignored, even if fail_on_unknown_attributes=True.

I may also add that sometimes xsi-attributes _are_ relevant.
The following:
```python
   def test_page_domain(self):
        xml = """<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
<api:pageSearchResult xmlns:api="urn:vpro:api:2013" xmlns="urn:vpro:media:2009" xmlns:media="urn:vpro:media:2009" xmlns:shared="urn:vpro:shared:2009" xmlns:pages="urn:vpro:pages:2013" total="432" totalQualifier="EQUAL_TO" offset="0" max="0">
  <api:items>
    <api:item xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="api:searchResultItem" score="0.3099519">
      <api:result xsi:type="pages:pageType" url="https://www.vprobroadcast.com/play~WO_VPRO_2297327~antibiotics~.html" type="VIDEO" creationDate="2019-11-22T14:15:09.443+01:00" lastModified="2019-11-22T14:15:09.443+01:00" lastPublished="2019-11-22T14:15:24.341+01:00" publishStart="2015-10-16T14:23:14.729+02:00" refCount="0" sortDate="2015-10-16T14:23:14.729+02:00">
        <pages:crid>crid://vpro/media/vprobroadcast/WO_VPRO_2297327</pages:crid>
        <pages:broadcaster id="VPRO"></pages:broadcaster>
        <pages:portal id="VPROBROADCAST" url="https://www.vprobroadcast.com">
          <pages:name>www.vprobroadcast.com</pages:name>
        </pages:portal>
        <pages:title>Antibiotics</pages:title>
        <pages:images>
          <pages:image type="PICTURE" url="https://images.poms.omroep.nl/image/s360/665086.jpg">
            <pages:title>Antibiotica</pages:title>
            <pages:description>Labyrint</pages:description>
          </pages:image>
        </pages:images>
      </api:result>
    </api:item>
  </api:items>
  <api:facets/>
  <api:selectedFacets/>
</api:pageSearchResult>
"""

        object = poms.from_string(xml) # FAILS: xsi:type not properly handled?
```
Fails with 
```
  
>           object = poms.from_string(xml) # FAILS: xsi:type not properly handled?

xsdata_xml_test.py:153: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Users/michiel/venvs/pyapi-env/lib/python3.11/site-packages/npoapi/data/poms.py:75: in from_string
    return parser.from_string(source)
/Users/michiel/venvs/pyapi-env/lib/python3.11/site-packages/xsdata/formats/bindings.py:25: in from_string
    return self.from_bytes(source.encode(), clazz)
/Users/michiel/venvs/pyapi-env/lib/python3.11/site-packages/xsdata/formats/bindings.py:29: in from_bytes
    return self.parse(io.BytesIO(source), clazz)
/Users/michiel/venvs/pyapi-env/lib/python3.11/site-packages/xsdata/formats/dataclass/parsers/bases.py:54: in parse
    result = handler.parse(source)
/Users/michiel/venvs/pyapi-env/lib/python3.11/site-packages/xsdata/formats/dataclass/parsers/handlers/lxml.py:52: in parse
    return self.process_context(ctx)
/Users/michiel/venvs/pyapi-env/lib/python3.11/site-packages/xsdata/formats/dataclass/parsers/handlers/lxml.py:67: in process_context
    self.parser.end(
/Users/michiel/venvs/pyapi-env/lib/python3.11/site-packages/xsdata/formats/dataclass/parsers/bases.py:152: in end
    return item.bind(qname, text, tail, objects)
/Users/michiel/venvs/pyapi-env/lib/python3.11/site-packages/xsdata/formats/dataclass/parsers/nodes/element.py:85: in bind
    self.bind_attrs(params)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <xsdata.formats.dataclass.parsers.nodes.element.ElementNode object at 0x120d5afd0>
params = {}

    def bind_attrs(self, params: Dict):
        """Parse the given element's attributes and any text content and return
        a dictionary of field names and values based on the given class
        metadata."""
    
        if not self.attrs:
            return
    
        for qname, value in self.attrs.items():
            var = self.meta.find_attribute(qname)
            if var and var.name not in params:
                self.bind_attr(params, var, value)
            else:
                var = self.meta.find_any_attributes(qname)
                if var:
                    self.bind_any_attr(params, var, qname, value)
                else:
                    if self.config.fail_on_unknown_attributes:
>                       raise ParserError(
                            f"Unknown attribute {self.meta.qname}:{qname}"
                        )
E                       xsdata.exceptions.ParserError: Unknown attribute {urn:vpro:api:2013}pageType:{http://www.w3.org/2001/XMLSchema-instance}type

/Users/michiel/venvs/pyapi-env/lib/python3.11/site-packages/xsdata/formats/dataclass/parsers/nodes/element.py:138: ParserError

Process finished with exit code 1
```

Which I think is incorrect too, and may be related. The difference is that in this case the xsi:type actually has a function.

--- a/.pre-commit-config.yaml
+++ b/.pre-commit-config.yaml
@@ -41,7 +41,7 @@ repos:
       - id: docformatter
         args: [ "--in-place", "--pre-summary-newline" ]
   - repo: https://github.com/pre-commit/mirrors-mypy
-    rev: v1.5.0
+    rev: v1.5.1
     hooks:
       - id: mypy
         files: ^(xsdata/)

--- a/tests/formats/dataclass/parsers/nodes/test_element.py
+++ b/tests/formats/dataclass/parsers/nodes/test_element.py
@@ -221,6 +221,15 @@ class ElementNodeTests(FactoryTestCase):
 
         self.assertEqual("Unknown attribute ExtendedType:a", str(cm.exception))
 
+    def test_bind_with_fail_on_unknown_attributes_ignores_xsi_attributes(self):
+        self.node.meta = self.context.build(ExtendedType)
+        self.node.config.fail_on_unknown_attributes = True
+        self.node.attrs = {QNames.XSI_TYPE: "b"}
+
+        objects = []
+        self.node.bind("foo", "text", "tail", objects)
+        self.assertEqual(1, len(objects))
+
     @mock.patch("xsdata.formats.dataclass.parsers.nodes.element.logger.warning")
     def test_bind_objects(self, mock_warning):
         self.node.meta = self.context.build(TypeC)

--- a/xsdata/formats/dataclass/parsers/nodes/element.py
+++ b/xsdata/formats/dataclass/parsers/nodes/element.py
@@ -17,6 +17,8 @@ from xsdata.formats.dataclass.parsers.utils import ParserUtils
 from xsdata.formats.dataclass.parsers.utils import PendingCollection
 from xsdata.logger import logger
 from xsdata.models.enums import DataType
+from xsdata.models.enums import Namespace
+from xsdata.utils.namespaces import target_uri
 
 
 class ElementNode(XmlNode):
@@ -134,7 +136,10 @@ class ElementNode(XmlNode):
                 if var:
                     self.bind_any_attr(params, var, qname, value)
                 else:
-                    if self.config.fail_on_unknown_attributes:
+                    if (
+                        self.config.fail_on_unknown_attributes
+                        and target_uri(qname) != Namespace.XSI.uri
+                    ):
                         raise ParserError(
                             f"Unknown attribute {self.meta.qname}:{qname}"
                         )

